### PR TITLE
♻️ Delete `size` arg from `StreamReader.feed_data`

### PR DESCRIPTION
**Pull request by dzhalaevd**
## What do these changes do?
Removed the unused `size` parameter from the `feed_data` method to simplify the code and usage of the method.

## Are there changes in behavior for the user?
If the user is using the feed_data method without the size parameter, there should be no changes in the behavior of the method for them. However, if the size parameter was used, there may be changes in the behavior of the method.

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

File: CHANGES/7265.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+The unused size parameter in the feed_data method has been removed
```
Comment (webknjaz) [inline CHANGES/7265.feature]: It's not a feature but a breaking change in the public API, probably.

File: aiohttp/streams.py [hunk 2]
```diff
@@ -228,7 +228,7 @@ def unread_data(self, data: bytes) -> None:
         self._eof_counter = 0
 
     # TODO: size is ignored, remove the param later
```
Comment (Dreamsorcerer) [inline aiohttp/streams.py]:
```suggestion
```

File: aiohttp/http_parser.py [hunk 3]
```diff
@@ -729,13 +729,13 @@ def feed_data(
...
+                self.payload.feed_data(chunk)
                 if self._length == 0:
                     self.payload.feed_eof()
                     return True, b""
             else:
                 self._length = 0
-                self.payload.feed_data(chunk[:required], required)
+                self.payload.feed_data(chunk[:required])
                 self.payload.feed_eof()
                 return True, chunk[required:]
```
Comment (Dreamsorcerer) [inline aiohttp/http_parser.py]: Maybe some further improvement to do this:
```
self._length = max(required - len(chunk), 0)
self.payload.feed_data(chunk[:required])
if self._length == 0:
    self.payload.feed_eof()
    return True, chunk[required:]
```

And remove the chunk_len variable on the line above.

File: aiohttp/http_parser.py [hunk 4]
```diff
@@ -783,11 +783,11 @@ def feed_data(
...
-                        self.payload.feed_data(chunk, chunk_len)
+                        self.payload.feed_data(chunk)
                         return False, b""
                     else:
                         self._chunk_size = 0
-                        self.payload.feed_data(chunk[:required], required)
+                        self.payload.feed_data(chunk[:required])
                         chunk = chunk[required:]
                         self._chunk = ChunkState.PARSE_CHUNKED_CHUNK_EOF
                         self.payload.end_http_chunk_receiving()
```
Comment (Dreamsorcerer) [inline aiohttp/http_parser.py]: Same again here:
```
self._chunk_size = max(required - len(chunk), 0)
self.payload.feed_data(chunk[:required])
if self._chunk_size:
    return False, b""

chunk = chunk[required:]
...
```

And another chunk_len variable can be removed from the line above.

**Comment by webknjaz** [comment]
It wouldn't hurt to write actual full commit messages and PR titles. When I get a notification about "something happened with todo" or "some improvement", those don't look important enough to compell me not to skip the notifications. I have hundreds of those and will choose things that look meaningful and related to what I care about.

File: aiohttp/http_parser.py [hunk 5]
```diff
@@ -779,18 +771,12 @@ def feed_data(
...
-                        self._chunk_size = required - chunk_len
-                        self.payload.feed_data(chunk, chunk_len)
+                    if self._chunk_size:
                         return False, b""
-                    else:
-                        self._chunk_size = 0
-                        self.payload.feed_data(chunk[:required], required)
-                        chunk = chunk[required:]
-                        self._chunk = ChunkState.PARSE_CHUNKED_CHUNK_EOF
-                        self.payload.end_http_chunk_receiving()
```
Comment (Dreamsorcerer) [inline aiohttp/http_parser.py:793]: I think these 2 lines are still needed.

File: aiohttp/http_parser.py [hunk 6]
```diff
@@ -779,18 +771,14 @@ def feed_data(
...
-                        return False, b""
-                    else:
-                        self._chunk_size = 0
-                        self.payload.feed_data(chunk[:required], required)
-                        chunk = chunk[required:]
+                    if self._chunk_size:
                         self._chunk = ChunkState.PARSE_CHUNKED_CHUNK_EOF
                         self.payload.end_http_chunk_receiving()
+                        return False, b""
+                    chunk = chunk[required:]
```
Comment (Dreamsorcerer) [inline aiohttp/http_parser.py]:
```suggestion
                        return False, b""
                    chunk = chunk[required:]
                    self._chunk = ChunkState.PARSE_CHUNKED_CHUNK_EOF
                    self.payload.end_http_chunk_receiving()
```

This was in the else condition before...

File: CHANGES/7265.feature (hunk 1 above)
Comment (dzhalaevd) [reply CHANGES/7265.feature]: So, I should use `7265.removal`?

**Comment by dzhalaevd** [comment]
CI also crashes due to the missing of the `size` arg and for some other reasons, however I do not know how to fix that. And I would really appreciate it if you could take care of it. Thanks!

**Comment by Dreamsorcerer** [comment]
Error seems pretty clear:
`aiohttp/http_parser.py:729: error: Missing positional argument "size" in call to "feed_data" of "DeflateBuffer"  [call-arg]`

Method is defined at: https://github.com/DavidRomanovizc/aiohttp/blob/master/aiohttp/http_parser.py#L854

**Comment by Dreamsorcerer** [comment]
Presumably, the same changes can be made, using `len()` instead of the `size` parameter.

File: tests/test_http_parser.py [hunk 7]
```diff
@@ -921,13 +921,14 @@ async def test_parse_eof_payload(self, stream: Any) -> None:
...
         assert out.is_eof()
-        assert [(bytearray(b"data"), 4)] == list(out._buffer)
+        assert [(b"data",)] == [(data,) for (data, _) in out._buffer]
 
     async def test_parse_no_body(self, stream: Any) -> None:
         out = aiohttp.FlowControlDataQueue(
             stream, 2**16, loop=asyncio.get_event_loop()
         )
         p = HttpPayloadParser(out, method="PUT")
+        p.feed_eof()
```
Comment (Dreamsorcerer) [inline tests/test_http_parser.py:931]: I'm unclear why these 2 need to change?

Comment (Dreamsorcerer) [inline tests/test_http_parser.py]: I'm unclear why these 2 need to change?

File: tests/test_streams.py [hunk 8]
```diff
@@ -1135,7 +1135,7 @@ def test_at_eof(self, buffer: Any) -> None:
 
     def test_feed_data(self, buffer: Any) -> None:
         item = object()
-        buffer.feed_data(item, 1)
+        buffer.feed_data(item)
         assert [(item, 1)] == list(buffer._buffer)
```
Comment (Dreamsorcerer) [inline tests/test_streams.py]: Looks like this becomes `0`, as the default behaviour appears to skip parsing (which is why the `object()` doesn't trigger any errors).

@webknjaz Do you think there's a chance this might break anything?

File: tests/test_http_parser.py (hunk 7 above)
Comment (dzhalaevd) [reply tests/test_http_parser.py]: This tests  have failed, and I thought, since we removed the `size` arg, then we need to little changed the assert in the test 🤔

Comment (Dreamsorcerer) [reply tests/test_http_parser.py]: Maybe, but I'd like to understand why these were not needed before. It's not obvious how these changes relate to a size parameter being removed, and could be an indication that some behaviour may break for users.

Comment (dzhalaevd) [reply tests/test_http_parser.py]: I can suggest that we have something wrong  in `HttpPayloadParser.feed_data`

Comment (Dreamsorcerer) [reply tests/test_http_parser.py]: Maybe, but what? We need to understand exactly why the change is needed.

Comment (Dreamsorcerer) [reply tests/test_http_parser.py]: Uhh, well, you've reverted the second change, and the test still seems to be passing. So, doesn't look like anything was broken anyway...

What about the first change from bytearray(), are you sure that's needed? If so, why?

Comment (dzhalaevd) [reply tests/test_http_parser.py]: So, earlier in `HttpPayloadParser.feed_data`, the `feed_data.payload` method was called which saved `chunk` and `len(chunk)`. Now that we have removed the `size` arg, we cannot save `len(chunk)` anymore. I believe to resolve this issue, we can modify the assertion like this:

```python
assert [bytearray(b"data")] == [item[0] for item in out._buffer]
```
(This is another option, not the one currently in my branch. If everything goes well, I can push it)

In two cases we compare data, but in old test we compared and len(data). But I think if the data somehow changes, then so does the size. 

Similarly, in `TestDataQueue.test_feed_data`, we need to replace the assertion with:

```python
assert [item] == [data[0] for data in buffer._buffer]
```

The reason is the same: we need to modify the assertion to compare only the first element of each tuple in `buffer._buffer` with the expected value of `item`.

Therefore, I am confident in changing the test. And also on my pc, all tests passed with these changes.

File: aiohttp/http_parser.py [hunk 9]
```diff
@@ -863,10 +851,11 @@ def __init__(self, out: StreamReader, encoding: Optional[str]) -> None:
...
         self.out.set_exception(exc)
 
-    def feed_data(self, chunk: bytes, size: int) -> None:
-        if not size:
+    def feed_data(self, chunk: bytes) -> None:
+        if not chunk:
             return
 
+        size = len(chunk)
         self.size += size
```
Comment (Dreamsorcerer) [inline aiohttp/http_parser.py]:
```suggestion
        self.size += len(chunk)
```

File: tests/test_http_parser.py (hunk 7 above)
Comment (Dreamsorcerer) [reply tests/test_http_parser.py]: I've looked into this, and that doesn't seem to be correct.
The buffer these tests refer to are `streams.DataQueue`, where the size parameter is still expected:
https://github.com/DavidRomanovizc/aiohttp/blob/master/aiohttp/streams.py#L559

So, maybe we want to drop the size parameter from that class as well? Then these changes can be reverted (only dropping the `4` from the test comparison).

File: tests/test_streams.py (hunk 8 above)
Comment (Dreamsorcerer) [reply tests/test_streams.py]: This is again because of the DataQueue. If that is updated, then this test should be fine as well.

File: aiohttp/streams.py [hunk 10]
```diff
@@ -635,10 +632,10 @@ def __init__(
         self._protocol = protocol
         self._limit = limit * 2
 
-    def feed_data(self, data: _T, size: int = 0) -> None:
-        super().feed_data(data, size)
+    def feed_data(self, data: _T) -> None:
+        super().feed_data(data)
 
-        if self._size > self._limit and not self._protocol._reading_paused:
+        if self._size < self._limit and not self._protocol._reading_paused:
```
Comment (Dreamsorcerer) [inline aiohttp/streams.py]: Why is this changed? Surely we don't want to pause reading if the size is still less than the limit...

File: aiohttp/streams.py [hunk 11]
```diff
@@ -580,9 +579,8 @@ def set_exception(self, exc: BaseException) -> None:
             self._waiter = None
             set_exception(waiter, exc)
 
-    def feed_data(self, data: _T, size: int = 0) -> None:
-        self._size += size
```
Comment (Dreamsorcerer) [inline aiohttp/streams.py:582]: Why is this removed? This will surely break the limit functionality. Just use `len(data)` to calculate the actual size?

File: aiohttp/streams.py [hunk 12]
```diff
@@ -608,8 +606,7 @@ async def read(self) -> _T:
                 raise
 
         if self._buffer:
-            data, size = self._buffer.popleft()
-            self._size -= size
```
Comment (Dreamsorcerer) [inline aiohttp/streams.py:610]: ^

Comment (dzhalaevd) [reply aiohttp/streams.py:610]: Because if we don't delete it we get error in almost all tests the `TypeError error: unsupported operand type(s) for -=: 'int' and 'StreamReader'`. In tests/test_client_functional.py and he refers to this block of code

File: aiohttp/streams.py (hunk 11 above)
Comment (dzhalaevd) [reply aiohttp/streams.py:582]: And if we don't delete this line, the `TestFlowControlDataQueue.test_resume_on_read` test will fall

File: aiohttp/streams.py (hunk 12 above)
Comment (Dreamsorcerer) [reply aiohttp/streams.py:610]: `len(data)`? Obviously you can't subtract a StreamReader from an int...

Though if it's possible for `data` to be something other than bytes, then maybe this won't work.

Comment (dzhalaevd) [reply aiohttp/streams.py:610]: OK, but we will still have tests falling in TestDataQueue. With TypeError error: object of type 'object' has no len(). Now it seems that we need to use `b"something"` instead of `object()` or handle error, but I'm not sure about that.

Comment (Dreamsorcerer) [reply aiohttp/streams.py:610]: Well, object() is probably unreasonable, so we could maybe update that test... I'll have a look through later.

Comment (dzhalaevd) [reply aiohttp/streams.py:610]: I added types checking to feed_data and now those tests don't crash, but I doubt the correctness of such a solution

File: aiohttp/streams.py (hunk 10 above)
Comment (dzhalaevd) [reply aiohttp/streams.py]: Under this condition, for some reason, the `test_feed_pause` test began falled

File: aiohttp/streams.py (hunk 12 above)
Comment (DavidRomanovizc) [reply aiohttp/streams.py:610]: @Dreamsorcerer could we resume work on this PR?
